classic_tetris_project/twitch.py

```diff
@@ -7,55 +7,50 @@
 
 from .env import env
 
-
-TWITCH_API = "https://api.twitch.tv/kraken/"
-TWITCH_API_HELIX = "https://api.twitch.tv/helix/"
+TWITCH_API = "https://api.twitch.tv/helix/"
 TWITCH_SERVER = "irc.chat.twitch.tv"
 TWITCH_PORT = 6667
 
 logger = logging.getLogger("twitch-bot")
 
 class APIClient:
-    def __init__(self, client_id):
+    def __init__(self, client_id, token):
         self.client_id = client_id
+        self.token =  token
         self.headers = {
-            "Accept": "application/vnd.twitchtv.v5+json",
-            "Client-ID": client_id
+            "Client-ID": client_id,
+            "Authorization": f"Bearer {token}"
         }
 
     def _request(self, endpoint, params={}, headers={}, api=TWITCH_API):
         response = requests.get(f"{api}{endpoint}", params=params,
                                 headers={**self.headers, **headers})
         return response.json()
 
-    def user_from_username(self, username, client=None):
-        response = self._request("users", { "login": username })
+    def user(self, client=None, **user_params):
+        params = {}
+
+        if ("login" in user_params):
+            params["login"] = user_params["login"]
+        if ("id" in user_params):
+            params["id"] = user_params["id"]
+
+        response = self._request("users", params)
+
         if "error" in response:
             return None
-        elif response["users"]:
-            user_obj = response["users"][0]
+        elif response["data"]:
+            user_obj = response["data"][0]
             return self.wrap_user_dict(user_obj, client)
-        else:
-            return None
+
+    def user_from_username(self, username=None, client=None):
+        return self.user(client, login=username)
 
     def user_from_id(self, user_id, client=None):
-        response = self._request(f"users/{user_id}")
-        if "error" in response:
-            return None
-        else:
-            return self.wrap_user_dict(response, client)
+        return self.user(client, id=user_id)
 
     def own_user(self, token, client=None):
-        user_data = self._request(f"users", api=TWITCH_API_HELIX,
-                                  headers={"Authorization": f"Bearer {token}"})
-        user_dict = user_data["data"][0]
-        return User(
-            client=client,
-            username=user_dict["login"],
-            id=user_dict["id"],
-            display_name=user_dict["display_name"],
-            tags=user_dict
-        )
```

Comment:
I think this method needs to stay somewhat the same. This is used for oauth login, so we need to use the token we received for the logged in user.

Reply:
Yeah good call - updated that here: https://github.com/professor-l/classic-tetris-project/pull/70/commits/c7e8537cf774e142e607d89e1ba4821b4b06640f

Probably good for that flow to stay separate anyway.